src/executables/util_sa.c

```diff
@@ -1739,7 +1739,12 @@ diagdb (UTIL_FUNCTION_ARG * arg)
 		  goto error_exit;
 		}
 	    }
-	  heap_dump_heap_file (thread_p, outfp, dump_records, class_name);
+	  if (heap_dump_heap_file (thread_p, outfp, dump_records, class_name) == ER_LC_UNKNOWN_CLASSNAME)
+	    {
+	      PRINT_AND_LOG_ERR_MSG (msgcat_message
+				     (MSGCAT_CATALOG_UTILS, MSGCAT_UTIL_SET_DIAGDB, DIAGDB_MSG_UNKNOWN_CLASS),
+				     class_name);
+	    }
```

Comment:
```suggestion
          error_code = heap_dump_heap_file ();
	  if (error_code != NO_ERROR)
	    {
	       if (error_code == ER_LC_UNKNOWN_CLASSNAME) // or using switch statement
                 {	      
	             PRINT_AND_LOG_ERR_MSG ();
	         }
	         
	       goto error_exit;	    
	     }
```

I think there should be more error cases which are not handled now.
How about make additional variable to get error code ?

Reply:
Other error codes have been decided not to be handled in this issue. However, I believe this change could improve handling of other error cases in the future.